rtengine/tmo_fattal02.cc

```diff
@@ -1242,15 +1244,33 @@ void ImProcFunctions::ToneMapFattal02(Imagefloat *rgb, const FattalToneMappingPa
                 float &r = rgb->r(y, x);
                 float &g = rgb->g(y, x);
                 float &b = rgb->b(y, x);
+                float s = 1.f;
                 if(l > 1.f) {
                     r = max(r * l - offset, r);
                     g = max(g * l - offset, g);
                     b = max(b * l - offset, b);
+                    if (satcontrol) {
+                        s = pow_F(1.f / l, 0.3f);
```

Comment:
Is there an explanation of how this function and the 0.3 value was chosen?

Reply:
This value 0.3, brings about the change in behavior when the values RGB  are very high. In the majority of cases this has no effect.
The change brings a reasonable weakening, we can see it on the TIF test image that I gave

Jacques

Reply:
Ok, I just wanted to know if it is empirical or has a logical derivation. It is the first, it seems.

Reply:
I think it is empirical. It is Alberto's work
But there are places in the code where I put similar functions which take into account that we are in the interval [0..1]. I sometimes used SQRT (0.5) or CBRT (0.33) because they can be faster.

Thank you again

Jacques